File: Questão12.go
package main

import "fmt"


type Produto struct {
    Nome  string
    Preco float64
}


func (p Produto) Combinar(outro Produto) Produto {
    return Produto{
        Nome:  p.Nome + " e " + outro.Nome,
        Preco: p.Preco + outro.Preco,
    }
}

func main() {
    produto1 := Produto{"Camiseta", 50.00}
    produto2 := Produto{"Calça", 100.00}
    produtoCombinado := produto1.Combinar(produto2)
    fmt.Printf("Produto combinado: %s - R$%.2f\n", produtoCombinado.Nome, produtoCombinado.Preco)
}

File: Questão12.java
class Produto {
    private String nome;
    private double preco;

    public Produto(String nome, double preco) {
        this.nome = nome;
        this.preco = preco;
    }

    public String getNome() {
        return nome;
    }

    public double getPreco() {
        return preco;
    }

    public Produto combinar(Produto outro) {
        return new Produto(this.nome + " e " + outro.nome, this.preco + outro.preco);
    }
}

public class Main {
    public static void main(String[] args) {
        Produto produto1 = new Produto("Camiseta", 50.00);
        Produto produto2 = new Produto("Calça", 100.00);
        Produto produtoCombinado = produto1.combinar(produto2);
        System.out.printf("Produto combinado: %s - R$%.2f%n", produtoCombinado.getNome(), produtoCombinado.getPreco());
    }
}
